test/tools/yulopti.cpp

```diff
@@ -171,28 +208,118 @@ class YulOpti
 		}
 	}
 
+	void objectApply(shared_ptr<Object> _object, function<void(Object&)> _fn)
+	{
+		for (auto const& subObjectNode: _object->subObjects) {
+			auto subObject = dynamic_pointer_cast<Object>(subObjectNode);
+
+			if (subObject != nullptr)
+				objectApply(subObject, _fn);
+		}
+
+		_fn(*_object);
+	}
+
+	void analyze(ErrorReporter& errorReporter)
```

Comment:
I think that the name of this function can be more explicit too. It is not clear what it is analyzing. Is is it the errors? Or the object source code?
Also, coding style.
```suggestion
	void analyze(ErrorReporter& _errorReporter)
```

Reply:
Renamed to `runCodeAnalyzer`

Reply:
Renamed to runCodeAnalyzer